sdk/src/main/java/com/ciscowebex/androidsdk/phone/internal/PhoneImpl.java

```diff
@@ -267,6 +273,33 @@ public void onEvent(@NonNull MercuryEvent event) {
         }
     }
 
+    private void checkActiveCalls(){
+        HashSet<String> mModels = new HashSet<>();
+        service.list(device, result -> {
+            List<LocusModel> models = result.getData();
+            if (models == null) {
+                Ln.d("Failure: " + result.getError());
+                return;
+            }
+            for (LocusModel model : models) {
+                mModels.add(model.getCallUrl());
+                doLocusEvent(model);
+            }
+
+            List<CallImpl> mCallsToCleanUp = new ArrayList<>();
+            for (CallImpl call : calls.values()) {
+                if(!mModels.contains(call.getUrl())){
+                    mCallsToCleanUp.add(call);
+                }
+            }
+
+            for(CallImpl call:mCallsToCleanUp){
+                call.end(new CallObserver.RemoteCancel(call));
```

Comment:
Shall we also remove this call from `calls` variable or it is taken care of when you call the `call.end` method?

Reply:
Yes. Its taken care as part of end method